Cannot open storage created with ert==8.4

Opening storage created with ert 8.4 fails with
```
>           params[data["name"]] = _KNOWN_PARAMETER_TYPES[param_type](**data)
E           TypeError: GenKwConfig.__init__() missing 1 required positional argument: 'update'
```

Can be reproduced with the following test:
https://github.com/equinor/ert/blob/dd13514d59d1920742a9180cb88109740a8d5d6c/tests/integration_tests/test_storage_migration.py#L11

Solving this issue should remove the xfail from that test.

When there is no GenKW you get the following error:

```
    @cached_property
    def parameter_configuration(self) -> Dict[str, ParameterConfig]:
        params = {}
        for data in self.parameter_info.values():
            param_type = data.pop("_ert_kind")
            if "update" not in data:
                data["update"] = True
>           params[data["name"]] = _KNOWN_PARAMETER_TYPES[param_type](**data)
E           KeyError: 'Field'

/src/ert/storage/local_experiment.py:195: KeyError
```

--- a/src/ert/storage/local_storage.py
+++ b/src/ert/storage/local_storage.py
@@ -288,6 +288,7 @@ class LocalStorage(BaseMode):
                 name=name,
             )
         )
+        self._index.version = _LOCAL_STORAGE_VERSION
         self._save_index()
 
     @require_write

--- a/src/ert/storage/migration/gen_kw.py
+++ b/src/ert/storage/migration/gen_kw.py
@@ -36,9 +36,3 @@ def migrate(path: Path) -> None:
                     del parameters_json[param_group][key_to_remove]
             with open(experiment / "parameter.json", "w", encoding="utf-8") as fout:
                 fout.write(json.dumps(parameters_json))
-
-    with open(path / "index.json", encoding="utf-8") as f:
-        index_json = json.load(f)
-    index_json["version"] = 2
-    with open(path / "index.json", "w", encoding="utf-8") as f:
-        f.write(json.dumps(index_json))

--- a/src/ert/storage/migration/observations.py
+++ b/src/ert/storage/migration/observations.py
@@ -1,6 +1,5 @@
 from __future__ import annotations
 
-import json
 from typing import TYPE_CHECKING
 
 from ert.storage.local_storage import local_storage_get_ert_config
@@ -19,9 +18,3 @@ def migrate(path: Path) -> None:
             output_path.mkdir(parents=True, exist_ok=True)
             for name, dataset in observations.items():
                 dataset.to_netcdf(output_path / f"{name}", engine="scipy")
-
-        with open(path / "index.json", encoding="utf-8") as f:
-            index_json = json.load(f)
-        index_json["version"] = 4
-        with open(path / "index.json", "w", encoding="utf-8") as f:
-            f.write(json.dumps(index_json))

--- a/src/ert/storage/migration/response_info.py
+++ b/src/ert/storage/migration/response_info.py
@@ -20,6 +20,5 @@ def migrate(path: Path) -> None:
             fout.write(json.dumps(response_info, default=str))
     with open(path / "index.json", encoding="utf-8") as f:
         index_json = json.load(f)
-    index_json["version"] = 3
     with open(path / "index.json", "w", encoding="utf-8") as f:
         f.write(json.dumps(index_json))

